DOC-8641: New query functions and changes to query functions

simon-dew opened the pull request:
The following draft documentation is ready for review:

* [OBJECT_FIELD][1], [OBJECT_PATHS][2], and [OBJECT_PAIRS_NESTED][3]
* [LPAD][4] and [RPAD][5]
* [DATE_TRUNC_MILLIS][6], [DATE_TRUNC_STR][7], and [STR_TO_MILLIS][8]

[1]: https://simon-dew.github.io/docs-site/DOC-8641/server/current/n1ql/n1ql-language-reference/objectfun.html#fn-obj-field
[2]: https://simon-dew.github.io/docs-site/DOC-8641/server/current/n1ql/n1ql-language-reference/objectfun.html#fn-obj-paths
[3]: https://simon-dew.github.io/docs-site/DOC-8641/server/current/n1ql/n1ql-language-reference/objectfun.html#fn-obj-pairs-nested
[4]: https://simon-dew.github.io/docs-site/DOC-8641/server/current/n1ql/n1ql-language-reference/stringfun.html#fn-str-lpad
[5]: https://simon-dew.github.io/docs-site/DOC-8641/server/current/n1ql/n1ql-language-reference/stringfun.html#fn-str-rpad
[6]: https://simon-dew.github.io/docs-site/DOC-8641/server/current/n1ql/n1ql-language-reference/datefun.html#fn-date-trunc-millis
[7]: https://simon-dew.github.io/docs-site/DOC-8641/server/current/n1ql/n1ql-language-reference/datefun.html#fn-date-trunc-str
[8]: https://simon-dew.github.io/docs-site/DOC-8641/server/current/n1ql/n1ql-language-reference/datefun.html#fn-date-str-to-millis

Docs issues: [DOC-8641](https://issues.couchbase.com/browse/DOC-8641) and [DOC-8806](https://issues.couchbase.com/browse/DOC-8806)

simon-dew committed:
DOC-8806: Add OBJECT_FIELD function

simon-dew committed:
Add OBJECT_PATHS function

simon-dew committed:
Add OBJECT_PAIRS_NESTED function

simon-dew committed:
Add LPAD function

simon-dew committed:
Add RPAD function

simon-dew committed:
Update formatting of Date and Time functions

simon-dew committed:
Update DATE_TRUNC_STR

simon-dew committed:
Update STR_TO_MILLIS

simon-dew committed:
Minor fixes

simon-dew commented:
Date formats to follow with DOC-8704

dhaggart reviewed (commented):
I think the statement in the example "Array subscripts and unique field names" for OBJECT_PATHS is incorrect and doesn't tally with the results - there is no array in the example statement.


For LPAD, RPAD you have the text "[Optional; default is whitespace, i.e. " "]"  but "whitespace" means a range of space characters; the default is Unicode U+0020 / ASCII 32.  So I would phrase that "[Optional; default is Unicode U+0020, i.e. " "]"  (or something like that).

-> But then of course what you have is consistent (albeit still wrong ;-)) with all the other functions/examples on the page.


In the components description for iso_week (https://simon-dew.github.io/docs-site/DOC-8641/server/current/n1ql/n1ql-language-reference/datefun.html#extracting-components) it appears "week" is missing formatting in the snippet "...different results between week and iso_week...".

simon-dew commented:
> I think the statement in the example "Array subscripts and unique field names" for OBJECT_PATHS is incorrect and doesn't tally with the results - there is no array in the example statement.

Fixed.

> For LPAD, RPAD you have the text "[Optional; default is whitespace, i.e. " "]" but "whitespace" means a range of space characters; the default is Unicode U+0020 / ASCII 32. So I would phrase that "[Optional; default is Unicode U+0020, i.e. " "]" (or something like that).
> 
> -> But then of course what you have is consistent (albeit still wrong ;-)) with all the other functions/examples on the page.

Fixed. Also updated LTRIM and RTRIM to specify that whitespace is `" \t\n\f\r"`.

> In the components description for iso_week (https://simon-dew.github.io/docs-site/DOC-8641/server/current/n1ql/n1ql-language-reference/datefun.html#extracting-components) it appears "week" is missing formatting in the snippet "...different results between week and iso_week...".

Fixed.

simon-dew committed:
Updates after review
